src/Database/Connection.php

```diff
@@ -60,9 +60,9 @@ class Connection implements ConnectionInterface
      * Driver object, responsible for creating the real connection
      * and provide specific SQL dialect.
      *
-     * @var \Cake\Database\DriverInterface
+     * @var array<string, \Cake\Database\DriverInterface>
      */
-    protected $_driver;
+    protected array $_drivers;
```

Comment:
If we use separate properties instead of an array we can type them in 5.x and potentially avoid the asserts() we have in `createDrivers()`.

Reply:
After changing everything to `getDriver()` this seems much simpler.